pkg/river/encoding/encoding.go

```diff
@@ -0,0 +1,139 @@
+package encoding
```

Comment:
API nit: we can greatly reduce the number of things that need to get exported here if the only exported method is 

```go 
func MarshalBodyJSON(input interface{}) ([]byte, error) 
``` 

Then in types from other packages you would use [`json.RawMessage`](https://pkg.go.dev/encoding/json#RawMessage) to expose the rendered JSON:

```go
// ComponentInfo represents a component in river.
type ComponentInfo struct {
	encoding.Field `json:",omitempty"`
	ID             string           `json:"id,omitempty"`
	Label          string           `json:"label,omitempty"`
	References     []string         `json:"referencesTo"`
	ReferencedBy   []string         `json:"referencedBy"`
	Health         *ComponentHealth `json:"health"`
	Original       string           `json:"original"`
	Arguments      json.RawMessage  `json:"arguments,omitempty"`
	Exports        json.RawMessage  `json:"exports,omitempty"`
	DebugInfo      json.RawMessage  `json:"debugInfo,omitempty"`
}
```

That would be a huge improvement over the exposed API and would keep things very tidy.

Reply:
bump

This is a nit for now, but I'd like for us to start thinking about the interfaces our packages expose.

Reply:
Reduced the scope significantly.